var/spack/repos/builtin/packages/dbcsr/package.py

```diff
@@ -66,6 +66,8 @@ class Dbcsr(CMakePackage, CudaPackage, ROCmPackage):
     # properties, since the parent class defines constraints for different archs
     # Instead just mark all unsupported cuda archs as conflicting.
     dbcsr_cuda_archs = ("35", "37", "60", "70")
+    with when("@2.3.0:"):  # Optimised A100 parameters only for @2.5.0:
+        dbcsr_cuda_archs = (*dbcsr_cuda_archs, "80")
```

Comment:
This is wrong. The context manager has effect only on directives, not on general Python assignments. To add values that are conditional to a constraint, see [here](https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-possible-values).

Reply:
Thanks for pointing it out @alalazo. It seemed to work on the local tests I did, so I thought it was OK.

If I understand correctly, `conditional` returns a `list` of `Value`s which makes the later `if arch not in dbcsr_cuda_archs` test fail.